qcloud_sms/models/qcloud_sms.py

```diff
@@ -0,0 +1,255 @@
+# Copyright 2018 Dinar Gabbasov <https://it-projects.info/team/GabbasovDinar>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+import logging
+
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from qcloudsms_py.httpclient import HTTPError
+    import phonenumbers
+except ImportError as err:
+    _logger.debug(err)
+
+
+class QCloudSMS(models.Model):
+    """Records with information about SMS messages."""
+
+    _name = 'qcloud.sms'
+    _description = 'SMS Messages'
+    _order = 'id desc'
+
+    STATE_SELECTION = [
+        ('draft', 'Draft'),
+        ('sent', 'Sent'),
+        ('done', 'Delivered'),
+        ('error', 'Error'),
+    ]
+
+    partner_ids = fields.Many2many('res.partner', string='Partner')
+    send_datetime = fields.Datetime(string='Sent', readonly=True, help='Date and Time of sending the message',
+                                    default=fields.Datetime.now)
+    message = fields.Text(string='Message', required=True)
+    state = fields.Selection(STATE_SELECTION, string='Status', readonly=True, default='draft',
+                             help='Status of the SMS message')
+    template_id = fields.Many2one('qcloud.sms.template', 'SMS Template')
+
+    def _phone_get_country(self, partner):
+        if 'country_id' in partner:
+            return partner.country_id
+        return self.env.user.company_id.country_id
```

Comment:
Why there is `phone` in method name?

Reply:
@yelizariev like this https://github.com/odoo/odoo/blob/11.0/addons/phone_validation/models/phone_validation_mixin.py#L11